### DISCUSSION TITLE
[Mobile] How to Bring Into View more than just a Focused Input?

### DISCUSSION THREAD

**Discussion by Soap-141** [unoplatform/uno#16181] [Q&A]
My team and I are working on an application that contains an account activation flow where you enter information and create a password, those information depends on the backend so we have a dynamic form control and we also have dynamic password rules in that control, when the form is very populated we have an input and a button under the password input with its rules.

If the password input is focused Uno does the proper thing and bring it into the view so we can see it, but we would like to add some kind of offset so it also bring the password rules into view (they are right under it).

Is there an existing way to do that?

Someone told me that a while back we had something to add offset to the bring into view behavior with `ScrollViewer` does this still exist?

**Comment by jeromelaban** [comment]
@MartinZikmund may know

**Comment by MartinZikmund** [comment]
@Soap-141 I think you should be able to adjust it by handling the [`UIElement.BringIntoViewRequested` event](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.bringintoviewrequestedeventargs?view=winrt-22621). I don't exactly remember the semantics, but it does have the offset properties and those should be able to handle your requirement.

**Reply by Soap-141** [reply]
Will try it out. Thanks!

**Reply by Soap-141** [reply]
@MartinZikmund Would you handle it for the Scroll Viewer itself?

**Reply by Soap-141** [reply]
Like this?
``` cs
private void OnLoaded(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
{
	this.TestScroll.BringIntoViewRequested += TestScroll_BringIntoViewRequested;
}

private void OnUnloaded(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
{
	this.TestScroll.BringIntoViewRequested -= TestScroll_BringIntoViewRequested;
}

private void TestScroll_BringIntoViewRequested(object sender, Microsoft.UI.Xaml.BringIntoViewRequestedEventArgs e)
{
	e.VerticalOffset = 200;
}
```

**Reply by MartinZikmund** [reply]
Yes something along those lines. The `BringIntoView` logic is ported from WinUI so technically as long as you find a setup that works there, it should also correctly work on Uno targets (and if not, that will be something for us to look into)

**Reply by Soap-141** [reply]
@MartinZikmund Doesn't look like it's doing anything, I've tried updating the values by handling the event and triggering it myself and it does nothing different. I've been testing on iOS because Android actually is working fine by default. And been working on 4.10.37 too.

**Reply by MartinZikmund** [reply]
Hmm, that's interesting... Can you please make some issue to track this? As BringIntoView should be supported fully, any discrepancy should be addressed

**Reply by Soap-141** [reply]
@MartinZikmund Sure I can do that, our budget for the project as been lowered and this issue pushed back, I will be able to make an issue in June.
